crates/trident_api/src/config/host/health.rs

```diff
@@ -0,0 +1,217 @@
+use serde::{Deserialize, Serialize};
+
+#[cfg(feature = "schemars")]
+use schemars::JsonSchema;
+
+use crate::config::host::scripts::{Script, ServicingTypeSelection};
+use crate::status::ServicingType;
+
+/// Configuration for the host OS health.
+#[derive(Serialize, Deserialize, Debug, Default, Clone, PartialEq, Eq)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+#[cfg_attr(feature = "schemars", derive(JsonSchema))]
+pub struct Health {
+    /// Checks to be run before Trident commits a serviced target OS as 'provisioned'. If any of
+    /// the checks fail, the commit will not be completed and, for A/B update, a rollback will be
+    /// triggered.
+    ///
+    /// These checks can run for installs and A/B updates. If `runOn` is specified for anything other
+    /// than 'clean-install' or 'ab-update' type, the check will be ignored. If 'all' is
+    /// specified, the check will run for both 'clean-install' and 'ab-update'.
+    ///
+    /// These checks are run in the target OS. The `$TARGET_ROOT` variable
+    /// will be set to '/' for consistency with postProvision scripts.
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub checks: Vec<Check>,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[cfg_attr(feature = "schemars", derive(JsonSchema))]
+pub enum Check {
+    /// Script that will be run. The success or failure of the script will define
+    /// the health of the target OS. The script will run where Trident is running,
+    /// either in the target OS or in a container running on the target OS (where
+    /// the target OS '/' is mounted as '/host').
+    Script(Script),
```

Comment:
scripts currently allow for a `path` source, but that will not work in commit

Reply:
adjusted code to fix